src/opcode.rs

```diff
@@ -1752,6 +1890,65 @@ opcode! {
 
 // === 6.7 ===
 
+opcode! {
+    /// Issue the equivalent of `pread(2)` with multi-shot semantics.
+    pub struct ReadMulti {
+        fd: { impl sealed::UseFixed },
+        len: { u32 },
+        buf_group: { u16 },
+        ;;
+        offset: u64 = 0,
+    }
+
+    pub const CODE = sys::IoringOp::ReadMultishot;
+
+    pub fn build(self) -> Entry {
+        let Self { fd, len, buf_group, offset } = self;
+
+        let mut sqe = sqe_zeroed();
+        sqe.opcode = Self::CODE;
+        assign_fd!(sqe.fd = fd);
+        sqe.off_or_addr2.off = offset;
+        sqe.len.len = len;
+        sqe.buf.buf_group = buf_group;
+        sqe.flags = IoringSqeFlags::BUFFER_SELECT;
+        Entry(sqe)
+    }
+}
+
+opcode! {
+    /// Issue the equivalent of a waitid(2) system call.
+    pub struct Waitid {
+        id: { types::WaitId },
+        infop: { *mut types::WaitIdStatus },
+        ;;
+        options: types::WaitIdOptions = types::WaitIdOptions::empty()
+    }
+
+    pub const CODE = sys::IoringOp::Waitid;
+
+    pub fn build(self) -> Entry {
+        use ::linux_raw_sys::general::{P_ALL, P_PGID, P_PID, P_PIDFD};
```

Comment:
Is this the only use of linux_raw_sys? We shouldn't introduce a new crate, instead first upstream the missing components to rustix and then use those types.